First version of zscore_matrix generation

TheSwagThief opened the pull request:
Compute z-score matrices from base matrices using mean and std connectivity matrices.

TheSwagThief committed:
First version of zscore_matrix generation

pep8speaks commented:
Hello @TheSwagThief, Thank you for updating !



* In the file [`onsetpy/io/tests/test_stats.py`](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py):

> [Line 14:1](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L14): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 25:30](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L25): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 26:29](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L26): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 36:34](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L36): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 37:33](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L37): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 45:34](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L45): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 46:26](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L46): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 54:34](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L54): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 55:33](https://github.com/Onset-lab/onsetpy/blob/2bf82af6ae0dcaf092cb25564d7e18dc6ba67c28/onsetpy/io/tests/test_stats.py#L55): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace





##### Comment last updated at 2024-11-04 19:31:30 UTC

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,96 @@
+    logging.getLogger().setLevel(logging.getLevelName(args.verbose))
+
+    assert_inputs_exist(parser, [args.mean, args.std] + args.base_matrices)
+    
```
Remove whitespace

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,96 @@
+    assert_matrices_compatible(parser, [mean_matrix, std_matrix] + base_matrices)
+
+    z_score_matrices = calculate_z_scores(mean_matrix, std_matrix, base_matrices)
+    
```
Remove whitespace

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,96 @@
+
+
+if __name__ == "__main__":
+    main()
```
Add new line in the end

TheSwagThief committed:
pep8 fixes

TheSwagThief committed:
pep8 fixes 2

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,100 @@
+)
+
+
+def calculate_z_scores(
```
This function should be place in the core of the repo (in the onsetpy folder). You could create a new folder in onsetpy named stats.py and put this function in. Ideally, I would like a unit test you can check in io for examples. I you don't know how to do it don't worry ! I will do it ! Anyway, I forgot to do the same thing for the first script.

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,100 @@
+        required=True,
+        help="Path to the standard deviation connectivity matrix in .npy format",
+    )
+    parser.add_argument(
```
I am not a fan to give multiple files in input when it is not required. Here you can just call n times the script for each matrix.

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,100 @@
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument(
+        "--mean", required=True, help="Path to the mean connectivity matrix in .npy format"
```
Why do you put mean and std as a required option ? You can just declare them as a mandatory argument (as base_matrices)

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,100 @@
+        "base_matrices", nargs="+", help="Paths to the base connectivity matrices in .npy format"
+    )
+    parser.add_argument(
+        "--out_prefix",
```
If you don't have multiple input files, you just need the output filename.

GuillaumeTh commented on scripts/onset_zscore_connectivity_matrix.py:
```diff
@@ -0,0 +1,100 @@
+
+    assert_inputs_exist(parser, [args.mean, args.std] + args.base_matrices)
+
+    mean_matrix = load_matrix(args.mean)
```
mean_matrix, std_matrix = load_matrix([args.mean, args.std])

GuillaumeTh reviewed (commented):
Very good job ! The code is good and the convention is respected ! I have some comments to address ! Do not hesitate if you want to discuss about the comments

TheSwagThief committed:
Added stats.py

Unsure if this is what I was supposed to do so let me know!

TheSwagThief committed:
Added test_zscore

Extremely unsure about this one, mostly copied the other tests, so let me know if it works and maybe we can talk about it anyways in person

TheSwagThief committed:
Update onset_zscore_connectivity_matrix.py according to Guillaume's recommendations

Everything SHOULD be sorted out, in theory :)

TheSwagThief committed:
Addressing my friend pep8s concerns

TheSwagThief committed:
extra pep8 concerns

TheSwagThief committed:
Moved stats.py

TheSwagThief committed:
Create test_stats.py

TheSwagThief committed:
Delete onsetpy/io/tests/test_zscore

TheSwagThief committed:
Delete onsetpy/stats.py

TheSwagThief committed:
Fixed test location

TheSwagThief committed:
Update stats.py for pep8

TheSwagThief committed:
Update test_stats.py pep8

TheSwagThief committed:
Update test_stats.py pep8 #2

TheSwagThief committed:
typo
